packages/aws-cdk-lib/aws-rds/adr/aurora-serverless-v2.md

```diff
@@ -0,0 +1,233 @@
+   */
+  readonly instances?: number;
+  /**
+   * @deprecated - use clusterInstances instead
+   */
+  readonly instanceProps?: InstanceProps;
+  readonly clusterInstances?: ClusterInstances;
+}
+
+interface ClusterInstances {
+  readonly writer: IClusterInstance;
+  readonly readers?: IClusterInstance[];
+}
+
+class ClusterInstance implements IClusterInstance {
+  public static provisioned(id: string, props: ProvisionedClusterInstanceProps = {}): IClusterInstance {
+    return new ClusterInstance(id, {
+      ...props,
+      instanceType: ClusterInstanceType.provisioned(props.instanceType),
+    });
+  }
+
+  public static serverlessV2(id: string, props: ServerlessV2ClusterInstanceProps = {}): IClusterInstance {
+    return new ClusterInstance(id, {
+      ...props,
+      instanceType: ClusterInstanceType.serverlessV2(),
+    });
+  }
+  private constructor(private id: string, private readonly props: ClusterInstanceProps = {}) { }
+
+  public bind(scope: Construct, cluster: DatabaseCluster, props: ClusterInstanceBindOptions): IAuroraClusterInstance {
+    // new class to represent this concept
+    return new AuroraClusterInstance(scope, this.id, {
+      cluster,
+      ...this.props,
+      ...props,
+    });
+  }
+}
+```
+
+_user configuration_
+```ts
+new rds.DatabaseCluster(this, 'Cluster', {
+  engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_03_0 }),
+  clusterInstances: {
+    writer: ClusterInstance.provisioned('writer'),
+    readers: [
+      // puts it in promition tier 0-1
+      ClusterInstance.serverlessV2('reader1', { scaleWithWriter: true }),
+      ClusterInstance.serverlessV2('reader2'),
+      ClusterInstance.serverlessV2('reader3'),
+    ],
+  },
+});
+```
+
+We can also provide instructions on how to migrate from `instanceProps` to
+`clusterInstances` which would just entail overwriting the logicalId of the
+instances. We could also add something like `ClusterInstance.fromInstanceProps(props: InstanceProps)`
+that allows customers to more easily migrate their existing configuration.
+
+## Alternatives
+
+The other alternative that was considered was to create an entirely new API
+(i.e. `rds.DatabaseClusterV2`). This was considered because it would allow us to
+create an API that was more tailored to the new functionality. Given all the
+context above, it was also originally unclear whether or not it would be possible
+to update the existing API while accounting for all the nuances with Serverless
+V2.
+
+Ultimately this was discarded because it looks like we will be able to account
+for the nuances and the benefit of allowing users to continue to use the
+existing API outweigh any ergonomic benefits of a completely new API.
+
+## Consequences
+
+The main consequence of this decision is maybe just some confusion on how
+to use the new properties vs the old ones and changing some property validation
+from linter to runtime validation (`instanceProps` was required and would show
```

Comment:
I think `@deprecation` and updating the examples in the README should be sufficient.